falcon/api.py

```diff
@@ -359,14 +391,11 @@ class can use suffixed responders to distinguish requests
         if '//' in uri_template:
             raise ValueError("uri_template may not contain '//'")
 
-        # NOTE(santeyio): This is a not very nice way to catch the suffix
-        # keyword. In python 3 it can be specified explicitly in the function
-        # definition, e.g.
-        # `add_route(self, uri_template, resource, *args, suffix=None, **kwargs)`
-        # but python 2 won't accept args like this.
-        suffix = kwargs.pop('suffix', None)
+        # NOTE(xgfone): we should allow the method map_http_methods to fix them,
+        # in order to prevent some arguments to be passed to the router.
+        args = list(args)
+        method_map = self.map_http_methods(resource, args, kwargs)
 
-        method_map = routing.map_http_methods(resource, suffix=suffix)
         routing.set_default_responders(method_map)
         self._router.add_route(uri_template, method_map, resource, *args,
```

Comment:
What do you think about making the new, override-able method a member of the router rather than `falcon.API`? That may give us a more cohesive design. All we would need to do is move the calls to get the mapping and to set the default responders into `add_route()`.

Reply:
Yes, you are right. I'm doing it, but it will break down the API `CompiledRouter.add_route()`, because the argument `method_map` is unnecessary.